storyruntime/Containers.py

```diff
@@ -139,6 +153,166 @@ def get_registry_url(cls, image):
+    async def check_memory(cls, story, line, service_name):
+        """
+        check_memory is utilized to retrieve the amount of memory
+        left for a specific service.
+
+        :param story: the current story
+        :param line: the current line
+        :param service_name: the name of the service you are checking
+        the memory for
+
+        :return: returns the amount of memory that is left for the
+         pod within the service's deployment, returns -1 if there
+         is no memory metrics available
+        """
+        app = story.app
+
+        container_name = cls.get_container_name(app, story.name,
+                                                line, service_name)
+        story.logger.debug(
+            f'Checking memory for '
+            f'the service {service_name} with '
+            f'the container {container_name}.'
+        )
+
+        res = await Kubernetes.make_k8s_call(
+            app.config, app.logger,
+            f'/apis/apps/v1/namespaces/{app.app_id}'
+            f'/deployments/{container_name}')
+
+        # if the container does not exist
+        # then it is in fact no live
+        if res.code != 200:
+            story.logger.debug(
+                f'Failed to check memory for the '
+                f'service {service_name} with the container '
+                f'{container_name}. Container does not exist.'
+            )
+            return -1
+
+        pod_metrics = await Kubernetes.get_pod_metrics(
+            app, container_name
+        )
+
+        if pod_metrics is None:
+            story.logger.debug(
+                f'Failed to check memory for the '
+                f'service {service_name} with the container '
+                f'{container_name}. No metrics available.'
+            )
+            return -1
+
+        res = await Kubernetes.make_k8s_call(
+            app.config, app.logger,
+            f'/apis/apps/v1/namespaces/{app.app_id}'
+            f'/deployments/{container_name}')
+
+        deployment_data = ujson.decode(res.body.decode('utf-8'))
+        deployment_spec = deployment_data['spec']['template']['spec']
+
+        container = deployment_spec['containers'][0]
+
+        limits = container['resources']['limits']
+
+        limits['memory'] = Kubernetes._memory_bytes(limits['memory'])
+        limits['cpu'] = Kubernetes._cpu_units(limits['cpu'])
+
+        story.logger.debug(f'Current metrics for the container '
+                           f'{container_name}: {pod_metrics}, '
+                           f'and current limits are set to: {limits}')
+
+        memory_left = limits['memory'] - pod_metrics['memory']
+
+        story.logger.debug(f'There is currently {memory_left} bytes '
+                           f'left of memory for {container_name}.')
+
+        if pod_metrics['memory'] >= limits['memory']:
+            story.logger.warn(
+                f'It looks like the service {service_name} '
+                f'is running out of memory!'
```

Comment:
```suggestion
                f'ran out of memory!'
```

You're already over the limit.